deepchecks/utils/typing.py

```diff
@@ -37,3 +38,25 @@ def __gt__(self, __value) -> bool:  # noqa: D105
 
     def __eq__(self, __value) -> bool:  # noqa: D105
         ...
+
+
+@runtime_checkable
+class BasicModel(Protocol):
+    """Traits of a model that are necessary for deepchecks."""
+
```

Comment:
Just wondering what would happen when we try to access feature_importance_.
it will simply calculate permutation importance?